src/languages/nix.js

```diff
@@ -9,85 +9,319 @@ Category: system
   };
-  const ESCAPED_DOLLAR = {
-    className: 'char.escape',
-    begin: /''\$/,
+
+  const PATH_PIECE = "[A-Za-z0-9_\\+\\.-]+"
+  const PATH = {
+    scope: 'symbol',
+    match: new RegExp(`(\\.\\.|\\.|~)?/(${PATH_PIECE})?(/${PATH_PIECE})*(?=[\\s;])`),
   };
+
+  const OPERATOR = {
+    scope: 'operator',
+    relevance: 0,
+    match: hljs.regex.either(...[
+      '==',
+      '=',
+      '\\+\\+',
+      '\\+',
+      '<=',
+      '<\\|',
+      '<',
+      '>=',
+      '>',
+      '->',
+      /(?<=(\s|[0-9\.]+))-/,
+      '//',
+      '/',
+      '!=',
+      '!',
+      '\\|\\|',
+      '\\|>',
+      '\\?',
+      '\\*',
+      '&&',
+    ]),
+  };
+
   const ATTRS = {
-    begin: /[a-zA-Z0-9-_]+(\s*=)/,
+    begin: new RegExp(`(?<=(^|\\{|;)\\s*)${IDENTIFIER_REGEX}(\\.${IDENTIFIER_REGEX})*\\s*=(?!=)`),
     returnBegin: true,
     relevance: 0,
     contains: [
       {
-        className: 'attr',
-        begin: /\S+/,
-        relevance: 0.2
+        scope: 'attr',
+        match: new RegExp(`(?<=(^|\\{|;)\\s*)${IDENTIFIER_REGEX}(\\.${IDENTIFIER_REGEX})*(?=\\s*=(?!=))`),
+        relevance: 0.2,
       }
-    ]
+    ],
+  };
+
+  const NORMAL_ESCAPED_DOLLAR = {
+    scope: 'char.escape',
+    match: /\\\$/,
+  };
+  const INDENTED_ESCAPED_DOLLAR = {
+    scope: 'char.escape',
+    match: /''\$/,
+  };
+  const ANTIQUOTE = {
+    scope: 'subst',
+    begin: /\$\{/,
+    end: /\}/,
+    keywords: KEYWORDS,
+  };
+  const ESCAPED_DOUBLEQUOTE = {
+    scope: 'char.escape',
+    match: /'''/,
+  };
+  const ESCAPED_LITERAL = {
+    scope: 'char.escape',
+    match: /\\(?!\$)./,
   };
   const STRING = {
-    className: 'string',
-    contains: [ ESCAPED_DOLLAR, ANTIQUOTE ],
+    scope: 'string',
     variants: [
       {
         begin: "''",
-        end: "''"
+        end: "''",
+        contains: [
+          INDENTED_ESCAPED_DOLLAR,
+          ANTIQUOTE,
+          ESCAPED_DOUBLEQUOTE,
+          ESCAPED_LITERAL,
+        ],
       },
       {
         begin: '"',
-        end: '"'
-      }
-    ]
+        end: '"',
+        contains: [
+          NORMAL_ESCAPED_DOLLAR,
+          ANTIQUOTE,
+          ESCAPED_LITERAL,
+        ],
+      },
+    ],
+  };
+
+  const URL = {
+    scope: 'link',
+    match: /https?:\/\/[A-Za-z0-9\-._~:/?#[\]@!$&'()*+,=]+/,
+  };
+
+  const FUNCTION_PARAMS = {
+    scope: 'params',
+    match: new RegExp(`${IDENTIFIER_REGEX}\\s*:(?=\\s)`),
   };
+
   const EXPRESSIONS = [
     hljs.NUMBER_MODE,
     hljs.HASH_COMMENT_MODE,
     hljs.C_BLOCK_COMMENT_MODE,
+    hljs.COMMENT(
+      /\/\*\*(?!\/)/,
+      /\*\//,
+      {
+        subLanguage: 'markdown',
+        relevance: 0
+      }
+    ),
+    BUILTINS,
     STRING,
-    ATTRS
+    URL,
+    LOOKUP_PATH,
+    PATH,
+    FUNCTION_PARAMS,
+    ATTRS,
+    OPERATOR,
   ];
+
   ANTIQUOTE.contains = EXPRESSIONS;
+
+  const REPL = [
+    {
+      scope: 'meta.prompt',
+      match: /(?<=^\s*)nix-repl> /
```

Comment:
You can use `^` to match the beginning of a line.

Reply:
The idea was to leave some leeway for quickly copypasting stuff, and forgetting to remove whitespace at the start of the line. Is it okay if I just let any initial whitespace be part of the scope as a temporary workaround for the lookbehind?

Reply:
Does the repl prompt itself not start at the beginning of the ilne?

Reply:
Usually yes, but I'd expect the prompt to be highlighted even if you happen to have a space or two before it. It's not a big deal, I could maybe readdress it once lookbehinds are available.

Reply:
Fixed